packages/design-system/src/components/N8nDatatable/Datatable.vue

```diff
@@ -0,0 +1,200 @@
+<script lang="ts">
+import { computed, defineComponent, PropType, ref, useCssModule } from 'vue';
+import {
+	DatatableColumn,
+	DatatableRow,
+	DatatableRowDataType,
+} from '@/components/N8nDatatable/mixins';
+import { getValueByPath } from '../../utils';
+import { useI18n } from '../../composables';
+import N8nSelect from '../N8nSelect';
+import N8nOption from '../N8nOption';
+import N8nPagination from '../N8nPagination';
+
+export default defineComponent({
```

Comment:
why defineComponent.. not `setup` maybe?

Reply:
The main difference between the two is that `defineComponent` + `setup()` will expose all the returned functions and variables when referencing the component using `:ref="someRef"`, while `<script setup>` needs them to be explicitly exposed using `expose`. 

For Design System components we probably should always expose all the properties and functions.